Lib/mastermind.rb

```diff
@@ -0,0 +1,58 @@
+class Mastermind
+  def show_instructions
+    puts 'Welcome to Mastermind! Your challenge is
+to guess the secret four digit code in under 10 tries.
+
+The code can include the following colors:
+Red(r), Green(g), Orange(o), Blue(b), Yellow(y) and Purple(p)
+Enter your guess by entering four color values from above, such as rgyb or pggy.
+
+If you have a correct color in a correct position, you get a white pin.
+If you have a correct color in a different position, you get a black pin
+press \'q\' to quit, or enter your first guess.'
+  end
+end
+
+class Code
+  attr_reader :answer
+
+  def generate_code
+    @colors = ["r", "g", "o", "b", "y", "p"]
+    answer = (1..4).map { @colors[rand(6)] }
+  end
+end
+
+class Response
+  attr_reader :player_input
+
+  def initialize(new_input)
+    @valid_guess = false
+    @player_input = new_input
+  end
+
+  def get_input
+    puts "Please enter a guess from the available colors: r, g, o, b, y, p"
+    @player_input = gets.chomp.downcase.gsub(/[\W]/, "")
```

Comment:
If the `Response` class is handling the input, shouldn't the actual input validation be handled here?

You could use a recursive guard statement like:

```ruby
    get_input if length_error || invalid_guess
```

then implement the `invalid_guess` method to check for an invalid color submission